src/routes/friend.js

```diff
@@ -0,0 +1,85 @@
+const express = require('express');
+const router = express.Router();
+const User = require("../databases/schema/users");
+
+//Send a friend request to a user and add the user to the senders list of pending requests
+//Incoming: user's object _id, username to search for
+//Outgoing: status 200/error will be empty
+router.put("/addFriend", async (request, response) => {
+    const {id, search} = request.body;
+    const userObj = await User.findById(id);
+    //search by username for the requestee, if not found, return an error
+    const friendReq = await User.findOne({username : search});
+    if(!friendReq) {
+        return response.status(404).send({error: "No user exists for that username!"});
+    }
+    await User.findByIdAndUpdate(friendReq._id, { $push: { requests: userObj.username } });
+    await User.findByIdAndUpdate(id, { $push: { pending: friendReq.username } });
+    return response.status(200).send({error: ""});
+});
+
+//Accept or reject a friend request, either way, remove the request from both sides
+//Incoming: user's object _id, username of the request sender, 1 to accept or 0 to reject
+//Outgoing: status 200/error will be empty
+router.put("/processRequest", async (request, response) => {
```

Comment:
`/update/` or `/process/`?

Reply:
You are processing the friend request, by deciding whether to accept or reject it. Since the request is deleted afterwards, you're really not updating anything.

Reply:
Got it. The naming was kinda odd to me since frontend would be requesting to `/friend/processrequest` about accepting/rejecting a request (not sending one). Would splitting this out to 2 seperate endpoints, `/friend/accept` and `/friend/reject` (no need to validate the `accept` parameter), be better?

Reply:
I feel like it's wasteful since it does literally the same thing, just doing one extra thing if accept == 1. If you have a better name that isn't acceptReject then I'd be down, but it should be one endpoint